components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -1334,108 +1227,46 @@ private void updateLocalAndOutboundAuthenticationConfiguration(int applicationId
-        } finally {
-            IdentityApplicationManagementUtil.closeStatement(storeUseTenantDomainInLocalSubjectIdStmt);
-        }
-
-        PreparedStatement storeUseUserstoreDomainInLocalSubjectIdStmt = null;
-        try {
-            storeUseUserstoreDomainInLocalSubjectIdStmt = connection
-                    .prepareStatement(UPDATE_BASIC_APPINFO_WITH_USE_USERSTORE_DOMAIN_LOCAL_SUBJECT_ID);
-            // IS_USE_USERSTORE_DIMAIN_LOCAL_SUBJECT_ID=? WHERE TENANT_ID= ? AND ID = ?
-            storeUseUserstoreDomainInLocalSubjectIdStmt.setString(1, localAndOutboundAuthConfig
-                    .isUseUserstoreDomainInLocalSubjectIdentifier() ? "1" : "0");
-            storeUseUserstoreDomainInLocalSubjectIdStmt.setInt(2, tenantID);
-            storeUseUserstoreDomainInLocalSubjectIdStmt.setInt(3, applicationId);
-            storeUseUserstoreDomainInLocalSubjectIdStmt.executeUpdate();
-        } finally {
-            IdentityApplicationManagementUtil.closeStatement(storeUseUserstoreDomainInLocalSubjectIdStmt);
-        }
-
-        PreparedStatement enableAuthzStmt = null;
-        try {
-            enableAuthzStmt = connection
-                    .prepareStatement(UPDATE_BASIC_APPINFO_WITH_ENABLE_AUTHORIZATION);
-            enableAuthzStmt.setString(1, localAndOutboundAuthConfig.isEnableAuthorization() ? "1" : "0");
-            enableAuthzStmt.setInt(2, tenantID);
-            enableAuthzStmt.setInt(3, applicationId);
-            enableAuthzStmt.executeUpdate();
-        } finally {
-            IdentityApplicationManagementUtil.closeStatement(enableAuthzStmt);
-        }
-
-        PreparedStatement storeSubjectClaimUri = null;
-        try {
-            storeSubjectClaimUri = connection
-                    .prepareStatement(UPDATE_BASIC_APPINFO_WITH_SUBJECT_CLAIM_URI);
-            // SUBJECT_CLAIM_URI=? WHERE TENANT_ID= ? AND ID = ?
-            storeSubjectClaimUri.setString(1, localAndOutboundAuthConfig.getSubjectClaimUri());
-            storeSubjectClaimUri.setInt(2, tenantID);
-            storeSubjectClaimUri.setInt(3, applicationId);
-            storeSubjectClaimUri.executeUpdate();
-        } finally {
-            IdentityApplicationManagementUtil.closeStatement(storeSubjectClaimUri);
-        }
-
         AuthenticationStep[] authSteps = localAndOutboundAuthConfig.getAuthenticationSteps();
-
-        if (authSteps == null || authSteps.length == 0) {
-            // if no authentication steps defined - it should be the default behavior.
-            localAndOutboundAuthConfig
-                    .setAuthenticationType(ApplicationConstants.AUTH_TYPE_DEFAULT);
-        }
+        PreparedStatement storeLocalAndOutboundConfigs = null;
 
         try {
-            if (localAndOutboundAuthConfig.getAuthenticationType() == null) {
-                // no authentication type defined - set to default.
+            if (authSteps == null || authSteps.length == 0 ||
```

Comment:
This if block can be outside the try catch block isn't it?
Any reason why we decide to move it inside the try catch block?

Reply:
There were two if blocks, one outside and one inside. I removed the outside one. No specific reason

Reply:
If there are no reasons lets move the if outside of the try catch